lib/fog/xen_server/connection.rb

```diff
@@ -5,8 +5,11 @@ module XenServer
     class Connection
       attr_reader :credentials
 
-      def initialize(host, timeout)
-        @factory = XMLRPC::Client.new(host, "/")
+      def initialize(host, port, use_ssl, timeout)
```

Comment:
Unfortunately i don't have the spare time to do so. =s

@wkoszek Do you think you could add this change yourself? I would be happy to bring this in if we make this change.